src/careamics/transforms/normalize.py

```diff
@@ -67,26 +85,41 @@ def __call__(
         Tuple[np.ndarray, Optional[np.ndarray]]
             Transformed patch and target.
         """
-        norm_patch = self._apply(patch)
-        norm_target = self._apply(target) if target is not None else None
+        norm_patch = np.zeros_like(patch, dtype=np.float32)
+        norm_target = (
+            np.zeros_like(target, dtype=np.float32) if target is not None else None
+        )
+
+        for i in range(patch.shape[0]):
```

Comment:
It would be better to use broadcasting to do the normalization:

```python
array = np.arange(3 * 4 * 4).reshape(3, 4, 4)

means = np.mean(array, axis=(1, 2))[:, np.newaxis, np.newaxis]
stds = np.std(array, axis=(1, 2))[:, np.newaxis, np.newaxis]

norm_array = (array - means) / stds
norm_array.mean(axis=(1, 2)), norm_array.std(axis=(1, 2))
```

Reply:
We just need to adjust the number of `np.newaxis` depending on whether there is a `Z` dim. We know this from the length of the shape.